pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java

```diff
@@ -1089,12 +1091,25 @@ protected void handleSubscribe(final CommandSubscribe subscribe) {
                 }
                 Optional<Map<String, String>> subscriptionProperties = SubscriptionOption.getPropertiesMap(
                         subscribe.getSubscriptionPropertiesList());
+
+                boolean createTopicIfDoesNotExist = forceTopicCreation || (isAuthorizedToCreateTopic
+                        && service.isAllowAutoTopicCreation(topicName.toString()));
                 service.getTopic(topicName.toString(), createTopicIfDoesNotExist)
                         .thenCompose(optTopic -> {
                             if (!optTopic.isPresent()) {
-                                return FutureUtil
-                                        .failedFuture(new TopicNotFoundException(
-                                                "Topic " + topicName + " does not exist"));
+                                if (isAuthorizedToCreateTopic) {
+                                    return FutureUtil
+                                            .failedFuture(new TopicNotFoundException(
+                                                    "Topic " + topicName + " does not exist"));
+                                } else {
+                                    String msg = "Topic to subscribe does not exists and the Client is not"
+                                            + " authorized to create topic";
+                                    log.warn("[{}] {} with role {}", remoteAddress, msg, getPrincipal());
+                                    consumers.remove(consumerId, consumerFuture);
+                                    ctx.writeAndFlush(Commands.newError(requestId, ServerError.AuthorizationError,
+                                            msg));
+                                    return null;
```

Comment:
I don't think returning `null` here will give the desired behavior unless we modify the `thenAccept` block. It seems like the current paradigm in this section of the code is to throw an exception, so I think it would be appropriate to do that here.

Reply:
Did you see https://github.com/apache/pulsar/pull/17411/files#diff-1e0e8195fb5ec5a6d79acbc7d859c025a9b711f94e6ab37c94439e99b3202e84L1187-L1198 lines? It seems we are returning null and writing errors to the channel. Nope?

Reply:
I just rebased it from master.